### PR TITLE
Zsh compsys

### PR DESCRIPTION
**Pull request by llua**

### CODE REVIEW

File: shell-completion/zsh/_sd_machines [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+local typeverb=list
 
-local -a _machines
-_machines=("${(fo)$(__get_machines)}")
-typeset -U _machines
-if [[ -n "$_machines" ]]; then
+zparseopts -D -E -a type t+:
+[[ -n $type[(r)i] ]] && typeverb=list-images
+
+_machines=(${${${(f)"$(_call_program systemd-machines machinectl --full --no-legend --no-pager $typeverb)"}:#No (machines|images).}%% *})
```
Comment (fsateler) [inline shell-completion/zsh/_sd_machines]: This seems unnecessary since f9b1947f9b9bcdf2b28953a40b1c537038c2a958 .

**Comment by llua** [comment]
I don't think the tests will complete.

File: shell-completion/zsh/_machinectl [hunk 2]
```diff
@@ -1,20 +1,5 @@
...
-    local -a _machines
-    _machines=("${(fo)$(__get_available_machines)}")
-    typeset -U _machines
-    if [[ -n "$_machines" ]]; then
-        _describe 'machines' _machines
-    else
-        _message 'no machines'
-    fi
-}
-
```
Comment (keszybz) [inline shell-completion/zsh/_machinectl:17]: That zsh completion syntax is impenetrable. I've written a bunch of it and every time I feel like it's been ROT13-ed.

Can you explain in the commit message what is changed and why? What is "-t i"?

File: shell-completion/zsh/_machinectl [hunk 3]
```diff
@@ -4,34 +4,41 @@
...
+    "shell:open an interactive shell in a container"
+    "enable:enable automatic container start at boot"
+    "disable:disable automatic container start at boot"
+    "poweroff:power off one or more VMs/containers"
+    "reboot:reboot one or more VMs/containers"
+    "terminate:terminate one or more VMs/containers"
+    "kill:send signal to process or a VM/container"
+    "copy-to:copy files from the host to a container"
+    "copy-from:copy files from a container to the host"
+    "bind:bind mount a path from the host into a container"
```
Comment (keszybz) [inline shell-completion/zsh/_machinectl:22]: This too should be two commits (at least). One for the trivial case changes, other for the changes in logic.

**Pull request closed without merging**